### PR TITLE
Change path strings to pathlib.Path objects

### LINKED ISSUE #3545: task: Move strings referring to file system paths to `pathlib.Path` objects
### Component

API Server / GraphQL, Git Integration, Python SDK

### Task Description

We use strings to refer to file system directories and files across the code base. These strings can be found in functions as well as in classes (data classes, Pydantic models, configurations), as properties. Some examples below:

```python
class MainSettings(BaseSettings):
    model_config = SettingsConfigDict(env_prefix="INFRAHUB_")
    docs_index_path: str = Field(
        default="/opt/infrahub/docs/build/search-index.json",
        description="Full path of saved json containing pre-indexed documentation",
    )
```

```python
class Worktree(BaseModel):
    identifier: str
    directory: str # Directory in the FS
    commit: str
    branch: Optional[str] = None
```

In an effort to consolidate the code base and make it consistent with using the `pathlib` module where appropriate, we should do the same for class properties referring to directories or files.

### PR DESCRIPTION
**Pull request by PhillSimonds**
This PR touches a fair bit of code, so it may be hard to keep up to date if left in the open state for a long time.

The config object takes a `config_file_name` attribute currently. I believe it would be best to convert this to a more appropriate name of `config_file` with a type of `PathLike[str]` -- this way a path object or a string can be passed in. I didn't do this because config is used in a lot of places in the code, and may also be used in a lot of places external to the code base -- so I wanted to get feedback on if this is worth doing at some point, if we should create another issue...etc.

Fixes #3545

### CODE REVIEW

**Comment by PhillSimonds** [comment]
It looks like this is failing on just Playwright (timeout issue), which seems to be the same for some other PRs so an issue unrelated to this PR? LMK if I need to do anything to resolve that.

File: backend/tests/integration/conftest.py [hunk 1]
```diff
@@ -49,11 +49,11 @@ async def load_infrastructure_schema(db: InfrahubDatabase):
...
     tmp_schema = branch_schema.duplicate()
 
-    for file_name in os.listdir(base_dir):
-        file_path = os.path.join(base_dir, file_name)
+    for file_name in base_dir.iterdir():
+        file_path = base_dir / file_name
 
-        if file_path.endswith((".yml", ".yaml")):
-            schema_txt = Path(file_path).read_text(encoding="utf-8")
+        if str(file_path).endswith((".yml", ".yaml")):
```
Comment (gmazoyer) [inline backend/tests/integration/conftest.py]: You could use `file_path.suffix in (".yml", ".yaml")`.

File: backend/tests/integration/git/test_git_repository.py [hunk 2]
```diff
@@ -34,11 +33,11 @@ async def load_infrastructure_schema(db: InfrahubDatabase):
...
     tmp_schema = branch_schema.duplicate()
 
-    for file_name in os.listdir(base_dir):
-        file_path = os.path.join(base_dir, file_name)
+    for file_name in base_dir.iterdir():
+        file_path = base_dir / file_name
 
-        if file_path.endswith((".yml", ".yaml")):
-            schema_txt = Path(file_path).read_text(encoding="utf-8")
+        if str(file_path).endswith((".yml", ".yaml")):
```
Comment (gmazoyer) [inline backend/tests/integration/git/test_git_repository.py]: `file_path.suffix`

**Review by gmazoyer — COMMENTED** [review]
I think you can ignore the failure in the E2E tests, they seemed to be completeley coming completely from something else (ie playwright timeouts).

**Comment by dgarros** [comment]
would be good to rebase this PR with the branch `release-1.1`, currently `develop` is a bit behind and recently we did some changes in `release-1.1` that could impact this PR. #5032

File: backend/tests/integration/conftest.py (hunk 1 above)
Comment (PhillSimonds) [reply backend/tests/integration/conftest.py]: Ahh awesome, thanks!

**Comment by PhillSimonds** [comment]
Ahh OK cool. I'll rebase onto release-1.1 and re-tool the PR.

**Comment by PhillSimonds** [comment]
OK this is re-based on `release-1.1`, re-targeted on the same, and should be good to go!

**Review by ajtmccarty — APPROVED** [review]
looks good to me @PhillSimonds 
thanks for working on this
feel free to merge

**Review by gmazoyer — APPROVED** [review]

**Pull request merged**